src/Native/Utils.js

```diff
@@ -414,8 +415,9 @@ function toString(v)
 		var output = '';
 		for (var i in v)
 		{
-			if (i === 'ctor') continue;
-			var str = toString(v[i]);
+			var val = v[i];
+			if (val === v.ctor) continue;
```

Comment:
Right. Hmm. One other option is to always specify `ctor` as a string, both on creation and on retrieval, but that's a much bigger undertaking.

`{'ctor': '_Tuple0'}`
`obj['ctor']`